scripts/runTypewriter.ps1

```diff
@@ -0,0 +1,150 @@
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter verbosity
+    Optional. Specifies the typewriter.exe verbosity.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter docs
+    Optional. Specifies the path to the root of the local documentation repo.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter outputMetadataFileName
+    Optional. Specifies the name of the output metadata file name.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter endpointVersion
+    Optional. Specifies the 'v1.0' or 'beta' endpoint name. The default is 'v1.0'.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter properties
+    Optional. Custom properties to be passed to the generator.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter transform
+    Optional. The path to the transform XSLT for cleaning up the metadata file.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter output
+    Optional. The output path of the generated files. Typewriter creates this directory if it doesn't exist.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+
+.Parameter generationMode
+    Optional. Specifies the typewriter.exe generation mode.
+    See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator#using-typewriter for more information.
+#>
+
+Param(
+    [parameter(Mandatory = $false)][string]$metadata,
+    [parameter(Mandatory = $false)][string]$language,
+    [parameter(Mandatory = $false)][string]$verbosity,
+    [parameter(Mandatory = $false)][string]$docs,
+    [parameter(Mandatory = $false)][string]$outputMetadataFileName,
+    [parameter(Mandatory = $false)][string]$endpointVersion,
+    [parameter(Mandatory = $false)][string]$properties,
+    [parameter(Mandatory = $false)][string]$transform,
+    [parameter(Mandatory = $false)][string]$output,
+    [parameter(Mandatory = $false)][string]$generationMode,
+    [parameter(Mandatory = $false)][bool]$cleanup = $true
+)
+
+# VARIABLES
+$gh_owner_and_repo = 'microsoftgraph/MSGraph-SDK-Code-Generator'    # Org and repo where typewriter.exe is released.
+$tempDir           = '.\temp'
+$LASTEXITCODE      = $null
+$typewriterZipDrop = "$tempDir\typewriter.zip"
+$typewriterFilesDir = "$tempDir\Release\*"
+$typewriter = Join-Path $tempDir "typewriter.exe"
+
+# Create our temporary working directory.
+if ((Test-Path -PathType Container -Path $tempDir)) {
+    # Remove-Item $tempDir -Force -Recurse
```

Comment:
should this be enabled so that when we leave this if/else block we are in exactly the same state regardless of creation or previous existence.

Reply:
That's how I originally had it set up, but then realized that if I want to run typewriter multiple times in a pipeline, I don't necessarily want to download and unzip again every time I run this script.

Reply:
that makes sense but you are running the risk of a dirty state causing an issue.  In a non-test environment cleaning it up may work out be more stable and not as horrible to keep purging.  Technically with the clean up flag being true you already do this.

---
In reply to: [385447267](https://github.com/microsoftgraph/msgraph-metadata/pull/10#discussion_r385447267) [](ancestors = 385447267)

Reply:
Even with the clean up flag, we can be dirty. The case where this could be a problem is if there is a fix in typewriter and typewriter isn't cleaned up, we may not download the fixed typewriter.
Pushed a fix